R/learn.R

```diff
@@ -192,16 +191,14 @@ legacy_new_dm <- function(tables, data_model) {
   segment <- data_model_tables$segment
   # would be logical NA otherwise, but if set, it is class `character`
   display <- as.character(data_model_tables$display)
-  filter <- new_filters()
   zoom <- new_zoom()
   key_tracker_zoom <- new_key_tracker_zoom()
 
-  # Legacy compatibility
-  pks$column <- as.list(pks$column)
-
   pks <-
     pks %>%
-    nest(pks = -table)
+    # Legacy compatibility
+    mutate(column = vctrs::vec_cast(column, list())) %>%
```

Comment:
Do we still need this?

Reply:
yes, I believe so, it's the difference of having a list of characters and a list of lists of characters in the `pks` column of the resulting `def`